topiary/src/tree_sitter.rs

```diff
@@ -84,6 +84,7 @@ pub fn apply_query(
     input_content: &str,
     query_content: &str,
     grammars: &[tree_sitter_facade::Language],
+    should_check_input_exhaustivity: bool,
```

Comment:
I wonder if it's possible to set this _only_ in the context of tests, so you don't have to pass it through as a parameter everywhere. Maybe there's a conditional compilation pragma that does this...

Reply:
I think the last version is much more reasonable: I expose `apply_query` for testing, instead of passing a boolean flag everywhere.